src/main/java/uk/gov/dwp/dataworks/guacamole/cognito/AuthenticationProviderService.java

```diff
@@ -0,0 +1,192 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class AuthenticationProviderService {
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationProviderService.class);
+
+    static final StringGuacamoleProperty KEYSTORE_URL = new StringGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "keystore-url";
+        }
+    };
+    static final BooleanGuacamoleProperty VALIDATE_ISSUER = new BooleanGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "validate-issuer";
+        }
+    };
+
+    static final StringGuacamoleProperty CLIENT_USERNAME = new StringGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "client-username";
+        }
+    };
+
+    static final StringGuacamoleProperty ISSUER = new StringGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "issuer";
+        }
+    };
+
+    static final StringGuacamoleProperty CLIENT_PARAMS = new StringGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "client-params";
+        }
+    };
+
+    static final StringGuacamoleProperty USERNAME_CLAIMTYPE = new StringGuacamoleProperty() {
+        @Override
+        public String getName() {
+            return "username-claimtype";
+        }
+    };
+
+    private String keystoreUrl;
+    private Boolean validateIssuer;
+    private String clientUsername;
+    private String clientParams;
+    private String issuer;
+
+    private JwkProvider cognito;
+
+    private final Environment environment;
+
+    @Inject
+    public AuthenticationProviderService(Environment environment) throws GuacamoleException, MalformedURLException, SigningKeyNotFoundException {
+
+        this.environment = environment;
+        keystoreUrl = environment.getRequiredProperty(KEYSTORE_URL);
+        validateIssuer = environment.getRequiredProperty(VALIDATE_ISSUER);
+        clientParams = environment.getRequiredProperty(CLIENT_PARAMS);
+        clientUsername = environment.getRequiredProperty(CLIENT_USERNAME);
+        issuer = environment.getRequiredProperty(ISSUER);
+
+        logger.info("Reading keystore from {}", keystoreUrl);
+        JwkProvider provider = new UrlJwkProvider( new URL(keystoreUrl));
+        logger.info("Read keystore");
+
+        cognito = new GuavaCachedJwkProvider(provider);
+        logger.info("Caching keystore contents");
+    }
+
+    public Map<String, GuacamoleConfiguration> getAuthorizedConfigurations(HttpServletRequest request) {
+
+        String token = request.getParameter("token");
+
+        logger.debug("Couldnt read token.");
+
+        if (token == null) {
+            return null;
+        }
+
+        logger.debug("Get jwt token {}", token);
+
+        // Decode token.
+        DecodedJWT decodedToken = JWT.decode(token);
+        Jwk algo = null;
```

Comment:
Why is this defined outside the `try` block below? I think it can be assigned straight from the below code

Reply:
Moved inside the try block